lib/improver/utilities/cube_manipulation_new.py

```diff
@@ -158,10 +97,179 @@ def strip_var_names(cubes):
     return cubes
 
 
+class ConcatenateCubes():
+    """
+    Class adding functionality to iris.concatenate_cubes().
+
+    Accounts for differences in attributes and allows promotion of scalar
+    coordinates to be associated with the dimension over which concatenation
+    is to be performed (eg can promote forecast_period to auxiliary for single
+    time point cube inputs).
+    """
+
+    def __init__(self, master_coord, coords_to_associate=None,
+                 coords_to_slice_over=None):
+        """
+        Initialise parameters
+
+        Args:
+            master_coord (str):
+                Coordinate to concatenate over.
+
+        Kwargs:
+            coords_to_associate (list):
+                List of coordinates to be associated with the master_coord.  If
+                master_coord is "time" this should be "forecast_reference_time"
+                OR "forecast_period", NOT both.
```

Comment:
We could ... but iris will raise quite an angry one if we are silly enough to try something like that.  The only reason I specified this in the docstring was because this was very wrong in the previous iteration of this function.

I should mention since a lot of your comments are on this function:

- I copied this with very little modification to be a class method rather than a standalone function (as concatenation was the only place it was called)
- I'm considering a recommendation to remove this function after the refactor, since as far as I can make out it's never called (it actually failed when given a "sensible" cube in its previous form, as I found in reworking unit tests), probably since we squeeze out scalar dimensions on the majority of input cubes on load.  If we're not using something like this we really shouldn't keep it.

I'll work through the review anyway but this is in the back of my mind.

Reply:
Could we add a unit test at the relevant point to demonstrate that this plugin will fail if given both forecast_reference_time and forecast_period. I know the error is being raised by Iris, but I would like to capture the expected behaviour here. If Iris is changed underneath us we will want to know that our expected behaviour here has also changed.

Reply:
OK, well this is weird.  Now it doesn't fail, it just gives a silly answer.
I'll put something in this function that makes it fail.

Reply:
Thanks for that, I'm glad we are catching it regardless of iris's behaviour.